### ISSUE TITLE
Document `type` property for datepicker component and update for screen readers

### ISSUE THREAD

**Issue by Spark450** [GovAlta/ui-components#2732]
# Description

- Document the new `type` property which has 2 options "input" or "calendar".
- And update the new `input` type to work properly with screen readers
- Add any missing code to ensure accessibility

## Acceptance criteria

- [x] Add a new `type` property in the Date Picker sandbox
- [x] Update the properties table with the new `type` property
- [x] include the following [date of birth example](https://www.figma.com/design/aIRjvBzpIUH0GbkffjbL04/%E2%9D%96-Patterns-library-%7C-DDD?node-id=1896-179590&t=HTIU4LpbTrXTs62a-1) in the "examples section of the website and add the included figma link
- [x] Add to `min` and `max` properties that they are only for `type="date"`
- [ ] Adjust the Datepicker `input` type to use proper aria-labels for the inputs
- [ ] Verify with the above change that the `input` type is usable by screen readers
  - Screen reader should read the following:
    - Focus month input: "[form item label], Select a month"
    - Focus day input: "Day, enter a 2-digit number"
    - Focus Year input: "Year, enter a four-digit number"

**Comment by Spark450** [comment]
@twjeffery Can you a look at the screen reader experience part of this issue?
